lms/services/grouping.py

```diff
@@ -97,6 +109,59 @@ def upsert_canvas_group(  # pylint: disable=too-many-arguments
             )
         )
 
+    def upsert_grouping_memberships(self, user: User, groups: List[Grouping]):
+        """
+        Upserts group memberships.
+
+        :param user:  User the that belongs to the groups
+        :param groups: List of groups the `user` belongs to
+        """
+        for group in groups:
+            if membership := (
+                self._db.query(GroupingMembership)
+                .filter_by(grouping_id=group.id, user_id=user.id)
+                .one_or_none()
+            ):
+                membership.updated = func.now()
+                continue
+
+            group.memberships.append(GroupingMembership(grouping=group, user=user))
+
+    def get_course_groupings_for_user(
```

Comment:
I had another version of this method that was more general (with an optional parent_id param so it worked to find memberships to top-level courses) but that mean that

- It needed and extra `application_instance` param 
- Having a `group_set_id` made even less sense than in this version